src/docfx/lib/html/HtmlAttribute.cs

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+
+namespace Microsoft.Docs.Build
+{
+    internal readonly struct HtmlAttribute
+    {
+        private readonly string _html;
+        private readonly (int start, int length) _nameRange;
+        private readonly (int start, int length) _valueRange;
+        private readonly (int start, int length) _tokenRange;
+
+        public (int start, int length) NameRange => _nameRange;
+
+        public (int start, int length) ValueRange => _valueRange;
+
+        public (int start, int length) TokenRange => _tokenRange;
+
+        public ReadOnlySpan<char> Name => _html.AsSpan().Slice(_nameRange.start, _nameRange.length);
```

Comment:
Can I think of `AsSpan()` is like creating a pointer to specific block of memory(here it's still allocated on heap?)

And can `AsSpan()` be called only once and cached, instead of many time?